core/src/orderbook/auction_data_reader.rs

```diff
@@ -53,7 +53,7 @@ impl AuctionDataReader {
         let index = self.index;
         let auction_elements = auction_elements
             .iter()
-            .filter(|x| x.in_auction(index) && x.order.sell_amount > 0);
+            .filter(|x| x.in_auction(index) && x.order.remaining > 0);
```

Comment:
Does the check also require the sell token balance to be `> 0`? I don't remember if the balance was somehow factored into the buy/sell amount calculations.

Reply:
This is a bit subtle, but we need to include orders even if they have sellTokenBalance of 0. Let's say the order is from token A->B. The solver could trade another order (C->A) of that user in the same batch to increase their sellTokenBalance of A. In that case we want the first order to still be in the instance so that the new proceeds of token A could potentially be traded right away into B.